engine/extensions-mondrian/src/main/java/org/pentaho/reporting/engine/classic/extensions/datasources/mondrian/DriverDataSourceCache.java

```diff
@@ -179,15 +179,12 @@ public boolean isWrapperFor( Class c ) {
     }
   }
 
-  private static LFUMap cache;
+  private static final LFUMap cache = new LFUMap( 20 );
 
   private DriverDataSourceCache() {
   }
 
-  public static synchronized DataSource createDataSource( final String jdbcUrl, final Properties properties ) {
-    if ( cache == null ) {
-      cache = new LFUMap( 20 );
-    }
+  public static DataSource createDataSource( final String jdbcUrl, final Properties properties ) {
```

Comment:
LFUMap is not threadsafe. This can blow up so I'd use a global lock here. If anyone creates datasources in a frequency where a global lock matters then they probably need punishment; or they should switch to the JNDI data sources and pooling (which is a form of punishment AFAIK)